### PR TITLE
[2.1.2] Adds `null` resilience to search $query

### PR DESCRIPTION
**Pull request by nunomaduro**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Need Doc update   | no

## Describe your change

PHP is not a strong typed language, and some developers may enter in the mistake of passing non strings into a `search` method. E.g: In the Laravel Framework:

```php
$query = request('query'); // This may be null.

$results = Post::search($query);
```

This Pull Request addresses this cases, the `null` type is now converted to an empty string.

### CODE REVIEW

**Comment by nunomaduro** [comment]
@julienbourdeau You are right! Just added some tests on this.

File: tests/Unit/RequestTestCase.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ protected function assertEndpointEquals(RequestInterface $request, $endpoint)
     protected function assertBodySubset($subset, RequestInterface $request)
     {
         $body = json_decode((string) $request->getBody(), true);
-        $this->assertArraySubset($subset, $body);
+        $this->assertArraySubset($subset, $body, true);
```
Comment (nunomaduro) [inline tests/Unit/RequestTestCase.php]: @julienbourdeau Nice improvement here, where this comparision is now `strict`.

File: tests/Unit/SearchTest.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        } catch (RequestException $e) {
+            $this->assertBodySubset(array('query' => '',), $e->getRequest());
+        }
+
+        $client = PlacesClient::create('id', 'key');
+
+        try {
+            $client->search(null);
+        } catch (RequestException $e) {
+            $this->assertBodySubset(array('query' => '',), $e->getRequest());
```
Comment (rayrutjes) [inline tests/Unit/SearchTest.php]: `array('query' => '',)` that is a weird syntax fomatting, expected?

Comment (nunomaduro) [reply tests/Unit/SearchTest.php]: @rayrutjes We are using the `symfony` cs preset, that rule seems to be on it.

**Pull request merged**